build-tools/packages/build-cli/src/library/packageExports.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import path from "node:path";
+
+import type { Logger, PackageJson } from "@fluidframework/build-tools";
+
+import { ApiTag } from "./apiTag";
+
+export interface ExportData {
+	/**
+	 * Location of file relative to package
+	 */
+	relPath: string;
+	/**
+	 * Conditions required for export access; in hierarchy order
+	 */
+	conditions: string[];
+	/**
+	 * Export is only .d.ts file
+	 */
+	isTypeOnly: boolean;
+}
+export type Node10CompatExportData = Omit<ExportData, "conditions">;
+
+/**
+ * Only the value types of exports that are records.
+ */
+type ExportsRecordValue = Exclude<Extract<PackageJson["exports"], object>, unknown[]>;
+
+function findTypesPathMatching(
+	mapQueryPathToApiTagLevel: Map<string | RegExp, ApiTag | undefined>,
+	exports: ExportsRecordValue,
+	conditions: string[],
+): (ExportData & { apiTagLevel: ApiTag | undefined }) | undefined {
+	for (const [entry, value] of Object.entries(exports)) {
+		if (typeof value === "string") {
+			if (entry === "types") {
+				for (const [key, apiTagLevel] of mapQueryPathToApiTagLevel.entries()) {
+					// eslint-disable-next-line max-depth
```

Comment:
Nit: extracting the body of the for-loop into its own function might make this disable unecessary.

Reply:
It would. I felt it was better to read the contents of the function here rather than extract for this aggressive rule (my opinion) and try to name/explain the function and handle the return. ?
We can also mitigate rule violation by making line 47 "if" negative and "continue"ing which would bring this all up one level, but continue can be harder to follow.

Reply:
I'm not particularly concerned about the line length rule itself, more commenting on the fact that there is a lot of logical nesting going on here. I'm wondering if it might make it a bit more readable to pull out pieces and document what they do in isolation. I'm honestly not sure if it would, but there is a fair amount going on here.

Reply:
I went in to see if comments would help clear things up and that lead me to the things that aren't handled as we don't author outside of a pattern. I started editing to handle those and found a way to move a little out and I think it is better now.